nearcore/src/state_sync.rs

```diff
@@ -304,6 +328,249 @@ async fn state_sync_dump(
     tracing::debug!(target: "state_sync_dump", shard_id, "Stopped state dump thread");
 }
 
+async fn get_missing_state_parts_for_epoch(
+    shard_id: ShardId,
+    chain_id: &String,
+    epoch_id: &EpochId,
+    epoch_height: u64,
+    total_parts: u64,
+    external: &ExternalConnection,
+) -> Result<Vec<u64>, anyhow::Error> {
+    let s3_directory =
```

Comment:
> `s3_directory`

That can also be a local filesystem directory. Please use a more generic name, such as `directory` or `external_storage_directory`.